### DISCUSSION TITLE
Add template blocks support to Markdown files

### DISCUSSION THREAD

**Discussion by ngustavo** [11ty/eleventy#4094] [Enhancement Queue]
### Is your feature request related to a problem? Please describe.

I've struggled so many times with creating an end-user-friendly setup with content blocks inside `index.md` that I think this could be someone else's issue too.

### Describe the solution you'd like

Use template blocks in Markdown files, e.g. Pug, Nunjucks, Liquid `block` syntax

I suggest using MDC-like syntax:

```mdx
# Blocks!
::motd
Hello World!
::
```

layout.njk
```njk
{{ content | safe }}
<p class="motd">
  {% block motd %} N/A {% endblock %}
</p>
```

output
```html
<h1>Blocks!</h1>
<p class="motd">Hello World!</p>
```

### Describe alternatives you've considered

Using front matter data passed to layouts by cascading. This is not optimal due to lack of syntax highlighting for markdown. These variables are intended for simple strings, not code-like data.

**Comment by pdehaan** [comment]
I haven't tried this in a while, but .md files are preprocessed by a template engine (default is LiquidJS). Per https://www.11ty.dev/docs/languages/markdown/

> _"**INFO:** Markdown files are by default pre-processed as Liquid templates. [You can change this default in your configuration file](https://www.11ty.dev/docs/config/#default-template-engine-for-markdown-files) (or disable it altogether). To change this for a single template and not globally, read [Changing a Template’s Rendering Engine](https://www.11ty.dev/docs/languages/)."_

So couldn't your .md files use [Liquid blocks](https://liquidjs.com/tags/layout.html) (or switch the template engine to Nunjucks if you prefer [Nunjucks blocks](https://mozilla.github.io/nunjucks/templating.html#block)). Although I think the caveat is that using blocks usually means you have to use `{% layout %}` tags instead of relying on Eleventy's layouts.

**Comment by zachleat** [comment]
Also—this seems related: https://github.com/11ty/eleventy/pull/2502

**Comment by zachleat** [comment]
> By default this repo closes enhancement request issues and places them in a queue for folks to upvote. We do this to avoid a large build-up of open issues. We got this idea from the [lodash project](https://twitter.com/samselikoff/status/991395669016436736).

> Don’t forget to upvote the first comment with 👍 to register your vote! You can view [the current enhancement backlog](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+).
